src/mgopurge/main.go

```diff
@@ -62,7 +63,11 @@ var allStages = []stage{
 		"prune",
 		"Prune finalised transactions",
 		func(db *mgo.Database, txns *mgo.Collection) error {
-			stats, err := jujutxn.CleanAndPrune(db, txns, -1)
+			stats, err := jujutxn.CleanAndPrune(jujutxn.CleanAndPruneArgs{
+				Txns: txns,
+				TxnsCount: -1,
```

Comment:
Why are you passing -1 here? The real count will be obtained if TxnCount is less than or equal to 0.

Part of the reason for the args struct was to allow optional args to be omitted in the call. I think it reads nicer if you leave TxnsCount out.